### PR TITLE
Support buckets containing underscore

### LINKED ISSUE #262: [BUG] GCS buckets with underscores are resolved as null
### Is this a possible security vulnerability?

- [X] This is NOT a possible security vulnerability

### Describe the bug

Underscores is a valid symbol in the bucket name, [the doc ](https://cloud.google.com/storage/docs/buckets#naming)

> Bucket names can only contain lowercase letters, numeric characters, dashes (-), underscores (_), and dots (.). Spaces are not allowed. Names containing dots require [verification](https://cloud.google.com/storage/docs/domain-name-verification).

The current approach uses  `java.net.URI`, which is not allowing to have underscores in the host name of URI. It leads to wrongly requested access boundaries, because the bucket name is being set as `null`.

### To Reproduce

```
|  Welcome to JShell -- Version 11.0.22
|  For an introduction type: /help intro

jshell> String location = "gs://test_bucket/iceberg/data"
location ==> "gs://test_bucket/iceberg/data"

jshell> URI uri = uri.create(location);
uri ==> gs://test_bucket/iceberg/data

jshell> uri.getHost()
$3 ==> null
```

### Actual Behavior

The bucket name is set as `null`

### Expected Behavior

The bucket name is set as `test_bucket`

### Additional context

I guess this [class](https://cloud.google.com/java/docs/reference/google-cloud-storage/latest/com.google.cloud.storage.BlobId#com_google_cloud_storage_BlobId_fromGsUtilUri_java_lang_String_) from `com.google.cloud.storage` can be used instead
```
import com.google.cloud.storage.BlobId;
...
BlobId blob = BlobId.fromGsUtilUri(location);
String bucket = blob.getBucket();
String path = blob.getName();
...
```

### System information

Object storage: GCS

### PR DESCRIPTION
**Pull request by eric-maynard**
# Description

Currently, we are using `URI.getHost` to extract the bucket from paths, which can return `null` unexpectedly when the bucket contains an underscore. [This seems to be](https://stackoverflow.com/questions/25993225/uri-gethost-returns-null-why) a [known issue](https://stackoverflow.com/questions/28568188/java-net-uri-get-host-with-underscores).

This PR introduces a thin new utility for extracting the path and adds a test for it.

Fixes #262 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation update
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Added a new suite, `StorageUtilTest`.

# Checklist:

Please delete options that are not relevant.

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

### CODE REVIEW

File: polaris-service/src/main/java/org/apache/polaris/service/storage/PolarisStorageIntegrationProviderImpl.java [hunk 1]
```diff
@@ -73,7 +73,7 @@ PolarisStorageIntegration<T> getStorageIntegrationForConfig(
                           HttpTransportFactory.class, NetHttpTransport::new));
         } catch (IOException e) {
           throw new RuntimeException(
-              "Error initializing default google credentials" + e.getMessage());
+              "Error initializing default google credentials. " + e.getMessage());
```
Comment (nk1506) [inline polaris-service/src/main/java/org/apache/polaris/service/storage/PolarisStorageIntegrationProviderImpl.java:76]: nit: not related to this fix.

**Review by nk1506 — APPROVED** [review]
LGTM.

File: polaris-core/src/test/java/org/apache/polaris/core/storage/StorageUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+public class StorageUtilTest {
+
+  @Test
+  public void testEmptyString() {
+    Assertions.assertThat(StorageUtil.getBucket("")).isNull();
+  }
+
+  @ParameterizedTest
+  @ValueSource(strings = {"s3", "gcs", "azure"})
```
Comment (nk1506) [inline polaris-core/src/test/java/org/apache/polaris/core/storage/StorageUtilTest.java:34]: nit: do you think here values should be inline with iceberg supported [scheme](https://github.com/apache/iceberg/blob/f7c6d57a03282516c6f988fa23a809416cd765b1/core/src/main/java/org/apache/iceberg/io/ResolvingFileIO.java#L59) ? something like
`s3, s3a, s3n, gs, abfs, abfss`

**Review by RussellSpitzer — APPROVED** [review]

**Comment by RussellSpitzer** [comment]
Thanks @eric-maynard and thanks @nk1506 for review

**Pull request merged**